src/wallet_sync.rs

```diff
@@ -607,20 +789,50 @@ impl Wallet {
     pub fn find_incomplete_coinswaps(
         &self,
         rpc: &Client,
-    ) -> Result<HashMap<Hash160, Vec<(ListUnspentResultEntry, &WalletSwapCoin)>>, Error> {
+    ) -> Result<
+        HashMap<
+            Hash160,
+            (
+                Vec<(ListUnspentResultEntry, &IncomingSwapCoin)>,
+                Vec<(ListUnspentResultEntry, &OutgoingSwapCoin)>,
+            ),
+        >,
+        Error,
+    > {
         rpc.call::<Value>("lockunspent", &[Value::Bool(true)])
             .map_err(|e| Error::Rpc(e))?;
 
-        let completed_coinswap_hashvalues = self
-            .swap_coins
+        let mut completed_coinswap_hashvalues = self
+            .incoming_swap_coins
             .values()
             .filter(|sc| sc.other_privkey.is_some())
             .map(|sc| read_hashvalue_from_contract(&sc.contract_redeemscript).unwrap())
             .collect::<HashSet<Hash160>>();
+        completed_coinswap_hashvalues.extend(
+            self.outgoing_swap_coins
+                .values()
+                .map(|sc| read_hashvalue_from_contract(&sc.contract_redeemscript).unwrap()),
+        );
```

Comment:
Remove these lines. They add all the hashvalues of every outgoing swap coin regardless of whether it is incomplete or not. It results in outgoing swap coins never being returned by this function, which is the cause of the bug I found where incomplete swapcoins are not shown in `wallet-balance`.

Reply:
Thanks for retesting and debugging. Fixed